Makefile

```diff
@@ -750,9 +751,18 @@ buildbox-grpc:
 		lib/multiplexer/test/ping.proto \
 		lib/web/envelope.proto
 
+	# proto files within the api module must be passed with the 'M' flag. This way
+	# protoc will generate files with the correct API_IMPORT_PATH in the case where
+	# the import path has a version suffix, e.g. github.com/gravitational/teleport/api/v8
+
 	protoc -I=.:$$PROTO_INCLUDE \
 		--proto_path=api/client/proto \
-		--gogofast_out=plugins=grpc:api/client/proto \
+		--gogofast_out=plugins=grpc,\
+Mgithub.com/gravitational/teleport/api/types/types.proto=$(API_IMPORT_PATH)/types,\
+Mgithub.com/gravitational/teleport/api/types/events/events.proto=$(API_IMPORT_PATH)/types/events,\
+Mgithub.com/gravitational/teleport/api/types/wrappers/wrappers.proto=$(API_IMPORT_PATH)/types/wrappers,\
+Mgithub.com/gravitational/teleport/api/types/webauthn/webauthn.proto=$(API_IMPORT_PATH)/types/webauthn:\
```

Comment:
Can we factor this out into a variable and pass all the module paths to all the protoc invocations in `api`? Or will protoc complain about unimported modules?

Reply:
Yeah that works, ptal